tests/unit/cmd/scan/test_precommit.py

```diff
@@ -0,0 +1,87 @@
+import os
+import subprocess
+from pathlib import Path
+
+import pytest
+
+from ggshield.cmd.secret.scan.precommit import (
+    check_is_merge_with_conflict,
+    check_is_merge_without_conflict,
+)
+from tests.repository import Repository
+
+
+@pytest.mark.parametrize(
+    "is_merge",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "with_conflict",
+    [True, False],
+)
+def test_is_merge(tmp_path, is_merge, with_conflict):
```

Comment:
I would split this test in 3:

When is_merge is False, without_conflict does not matter, so the is_merge=False part is executed twice.

The common code between with and without conflict is small enough that it can be duplicated. This common code can even be shortened a bit more: the test does not use "inital.md", so you don't need to create it: `create_commit()` uses `--allow-empty`.